text_extensions_for_pandas/array/test_tensor.py

```diff
@@ -207,6 +207,25 @@ def test_slice(self):
         expected = np.array([[3, 4], [5, 6]])
         npt.assert_array_equal(expected, result)
 
+    def test_bool_indexing(self):
+        s = TensorArray([[1, 2], [3, 4]])
+
+        result = s[[True, True]]
+        expected = np.array([[1, 2], [3, 4]])
```

Comment:
Boolean indexing should return a TensorArray, just like indexing with a slice does.

Reply:
it does, but I should add a check for the type as well
```
In [43]: arr = tp.TensorArray([[1, 2], [3, 4]])                                                                              

In [44]: type(arr[[True, True]])                                                                                             
Out[44]: text_extensions_for_pandas.array.tensor.TensorArray
```

Reply:
Any selection using a Boolean mask should return a TensorArray, including cases where the mask contains zero or one `True` values. Pandas follows those semantics for all existing types. For example:
``` python
>>> s = pd.Series([1, 2])
>>> type(s[[True, False]])
pandas.core.series.Series
>>> type(s.values[[True, False]])
numpy.ndarray
```
The same applies to lists of indexes, even of those lists contain exactly zero or one elements:
``` python
>>> type(s[[0]])
pandas.core.series.Series
```